Assets/MixedRealityToolkit/_Core/Inspectors/Profiles/MixedRealityControllerMappingProfileInspector.cs

```diff
@@ -236,500 +139,242 @@ public override void OnInspectorGUI()
-                            {
-                                invertAxesEnum = InvertAxesEnum.Y;
-                            }
-
-                            var modifedValue = (InvertAxesEnum)EditorGUILayout.EnumFlagsField(InvertAxes, invertAxesEnum, GUILayout.Width(customLabelWidth));
-
-                            invertXAxis.boolValue = (modifedValue & InvertAxesEnum.X) > 0;
-                            invertYAxis.boolValue = (modifedValue & InvertAxesEnum.Y) > 0;
-
-                            EditorGUIUtility.labelWidth = defaultLabelWidth;
-                            EditorGUIUtility.fieldWidth = defaultFieldWidth;
+                            controllerHandedness.intValue = 0;
                         }
-                        else
+
+                        switch (currentGenericType)
                         {
-                            EditorGUILayout.LabelField(GUIContent.none, GUILayout.Width(customLabelWidth));
+                            case 0:
+                                thisProfile.MixedRealityControllerMappingProfiles[controllerList.arraySize - 1].ControllerType.Type = typeof(GenericUnityController);
+                                break;
+                            case 1:
+                                thisProfile.MixedRealityControllerMappingProfiles[controllerList.arraySize - 1].ControllerType.Type = typeof(GenericOpenVRController);
+                                break;
                         }
-                    }
 
-                    if ((AxisType)axisType.intValue == AxisType.SingleAxis ||
-                        (AxisType)axisType.intValue == AxisType.DualAxis)
-                    {
-                        var axisCodeX = interaction.FindPropertyRelative("axisCodeX");
-                        RenderAxisPopup(axisCodeX, customLabelWidth);
-                    }
-                    else
-                    {
-                        EditorGUILayout.LabelField(GUIContent.none, GUILayout.Width(customLabelWidth));
+                        // Only allow custom interaction mappings on generic controller types
+                        controllerType = thisProfile.MixedRealityControllerMappingProfiles[i].ControllerType.Type;
+                        useCustomInteractionMappings.boolValue = controllerType == typeof(GenericUnityController) ||
+                                                                 controllerType == typeof(GenericOpenVRController);
+                        interactionsList.ClearArray();
+                        serializedObject.ApplyModifiedProperties();
+                        thisProfile.MixedRealityControllerMappingProfiles[i].SetDefaultInteractionMapping(true);
+                        serializedObject.ApplyModifiedProperties();
                     }
 
-                    if ((AxisType)axisType.intValue == AxisType.DualAxis)
+                    if (renderControllerModels && controllerHandedness.intValue != 0)
```

Comment:
Why can controllers without a handedness not have a model?

Reply:
Because they're not spatial controllers, which won't be rendered in the scene.